### DISCUSSION TITLE
Feature Request: Simplify Signal-CLI into a TypeScript App

### DISCUSSION THREAD

**Discussion by kranebitt** [AsamK/signal-cli#1397] [Ideas]
**Description:**

Currently, Signal-CLI functions as a Git repository for a Signal Messenger command-line interface. To enhance usability and streamline the application, I propose simplifying the script into a TypeScript-based application. This transition will facilitate easier maintenance, improve readability, and potentially attract a wider range of contributors.

**Feature Details:**

- TypeScript Conversion: Restructure the existing script to utilize TypeScript for better code organization and type safety.
- Simplified Workflow: Create a more straightforward and intuitive workflow for users to navigate the Signal Messenger command-line interface.
- Improved Documentation: Update and enhance documentation to aid developers in understanding and contributing to the codebase.

**Benefits:**

- Enhanced Readability: TypeScript's static typing and structure will improve code readability and maintainability.
- Community Engagement: Simplification may attract more contributors, fostering a more active and diverse development community.
- User-Friendly: A streamlined interface will benefit users by providing a more intuitive Signal Messenger command-line experience.

Potential Implementation Steps:

- Analyze the existing codebase and plan the TypeScript migration strategy.
- Refactor code modules into TypeScript-compatible components.
- Test the functionality thoroughly to ensure feature parity and stability.
- Update documentation to reflect changes and guide contributors effectively.

Additional Information:

This enhancement aligns with the ongoing industry trend towards utilizing TypeScript for its robustness and maintainability. I am eager to contribute to this effort and would appreciate any guidance or support in implementing this feature.

Thank you for considering this proposal. I look forward to contributing to the advancement of Signal-CLI.

Related Issue(s): just a minor one, but this idea could not be implemented because a TypeScript version of Signal-CLI was not yet available: https://github.com/lowercasename/gathio/issues/121

Contributor: kranebitt

**Comment by AsamK** [comment]
Not going to happen. signal-cli is based on libsignal-service-java, which is a java library.
